apps/src/puzzleRatingUtils.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * @overview A helper class for all actions associated with Puzzle
+ * Ratings, aka the Fun-O-Meter.
+ */
+
+var PuzzleRatingUtils = {};
+
+module.exports = PuzzleRatingUtils;
+
+var dom = require('./dom');
+
+/**
+ * Construct the puzzle rating buttons themselves
+ *
+ * @returns {Element} div containing puzzle ratng buttons with attached
+ *          click handlers
+ */
+PuzzleRatingUtils.buildPuzzleRatingButtons = function () {
+  var buttonContainer = document.createElement('div');
+  buttonContainer.id = 'puzzleRatingButtons';
+  buttonContainer.innerHTML = require('./templates/puzzleRating.html.ejs')();
+
+  var buttons = buttonContainer.querySelectorAll('.puzzle-rating-btn');
+  var buttonClickHandler = function () {
+    for (var i = 0, button; (button = buttons[i]); i++) {
+      if (button != this) {
+        $(button).removeClass('enabled');
+      }
+    }
+    $(this).toggleClass('enabled');
+  };
+  for (var i = 0, button; (button = buttons[i]); i++) {
+    dom.addClickTouchEvent(button, buttonClickHandler);
+  }
+
+  return buttonContainer;
+};
```

Comment:
Mostly because it's such a simple little element and I wasn't positive that we had react support everywhere that this is going to appear (which is ... everywhere).

If we do, I'd be more than happy to swap it over.